redfish-core/lib/oem/ibm/usb_code_update.hpp

```diff
@@ -0,0 +1,140 @@
+#pragma once
+
+#include "async_resp.hpp"
+#include "dbus_utility.hpp"
+#include "redfish_util.hpp"
+
+#include <variant>
+
+namespace redfish
+{
+
+static constexpr const char* usbCodeUpdateObjectPath =
+    "/xyz/openbmc_project/control/service/phosphor_2dusb_2dcode_2dupdate";
+
+/**
+ * @brief Get the service name of the path
+ *
+ * @param[in] aResp     Shared pointer for generating response message.
+ * @param[in] path      The D-Bus Object path
+ * @param[in] handler   Call back
+ *
+ * @return None.
+ */
+template <typename Handler>
+inline void getServiceName(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
+                           const std::string& path, Handler&& handler)
+{
+    // Map of service name to list of interfaces
+    using MapperServiceMap =

```

Comment:
isn't this defined already, see https://gerrit.openbmc-project.xyz/c/openbmc/bmcweb/+/47776/8/redfish-core/lib/systems.hpp

Reply:
> 

Which line?

Reply:
This line 

using ipsPropertiesType =
    std::vector<std::pair<std::string, dbus::utility::DbusVariantType>>;

Reply:
> This line
> 
> using ipsPropertiesType = std::vector<std::pair<std::string, dbus::utility::DbusVariantType>>;

They are different files.